obsproject/obs-studio
# Preparations for Portals + PipeWire camera, part 2

### Description

This is the second batch of commits from https://github.com/obsproject/obs-studio/pull/6669, rebased on top of `master` and adjusted to the most recent changes.

The biggest change that this round of preparations introduce is making the PipeWire code able to handle async sources. There are no actual async sources that use PipeWire, but this is such a big change that I reasoned it's better reviewed independently from the next - and even bigger - reworks.

There should be no functional changes to the screencapture code.

### Motivation and Context

I don't want code reviewers crying in a corner when reviewing the PipeWire camera merge request, so let's go incremental and chop choppy and handle this massive amount of code in digestible chunks :slightly_smiling_face: 

### How Has This Been Tested?

- Run OBS Studio on Linux
- Add a PipeWire capture of any kind
- It should continue to work as it used to before

### Types of changes



 - Tweak (non-breaking change to improve existing functionality)

 - Code cleanup (non-breaking change which makes code smaller or more readable)



### Checklist:
